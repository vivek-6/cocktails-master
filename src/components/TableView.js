import React from "react";
import capitalize from "lodash/capitalize";
import { withStyles } from "@material-ui/core/styles";
import ConditionalHidden from "./ConditionalHidden";
import CocktailActions from "./CocktailActions";
import {
  TableRow,
  TableHead,
  TableContainer,
  TableCell,
  TableBody,
  Table
} from "@material-ui/core";
import CocktailAvatar from "./CocktailAvatar";

const styles = () => {
  return {
    root: {
      width: "100%"
    }
  };
};

const TableView = ({ classes, displayedCocktails }) => {
  const columns = [
    {
      name: "name"
    },
    { name: "category", hideOnXS: true }
  ];

  return (
    <TableContainer className={classes.container}>
      <Table>
        <TableHead>
          <TableRow>
            <TableCell>Appearance</TableCell>
            {columns.map(column => (
              <ConditionalHidden key={column.name} hideOnXS={column.hideOnXS}>
                <TableCell>{capitalize(column.name)}</TableCell>
              </ConditionalHidden>
            ))}
            <TableCell align="right">Actions</TableCell>
          </TableRow>
        </TableHead>
        <TableBody>
          {displayedCocktails.map(cocktail => (
            <TableRow hover tabIndex={-1} key={cocktail.name}>
              <TableCell>
                <CocktailAvatar cocktail={cocktail} />
              </TableCell>
              {columns.map(column => {
                return (
                  <ConditionalHidden
                    key={column.name}
                    hideOnXS={column.hideOnXS}
                  >
                    <TableCell>{cocktail[column.name]}</TableCell>
                  </ConditionalHidden>
                );
              })}
              <TableCell align="right">
                <CocktailActions cocktail={cocktail} hideLabelOnXS={true} />
              </TableCell>
            </TableRow>
          ))}
        </TableBody>
      </Table>
    </TableContainer>
  );
};

export default withStyles(styles)(TableView);
