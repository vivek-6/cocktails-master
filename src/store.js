import appReducer from "./reducers";
import { createStore, applyMiddleware, compose } from "redux";

import {
  persistCurrentState,
  supportsPersistence
} from "./utilities/persistence";
import throttle from "lodash/throttle";

import thunk from "redux-thunk";

const devtools =
  window.__REDUX_DEVTOOLS_EXTENSION__ && window.__REDUX_DEVTOOLS_EXTENSION__();

const middlewares = compose(
  applyMiddleware(thunk),
  devtools || (a => a)
);

const store = createStore(appReducer, middlewares);

// persist parts of the store whenever it changes.
if (supportsPersistence()) {
  store.subscribe(
    throttle(() => {
      persistCurrentState(store.getState(), ["bar", "settings", "favourites"]);
    })
  );
}

export default store;
