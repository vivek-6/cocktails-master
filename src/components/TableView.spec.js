import React from "react";
import renderer from "react-test-renderer";
import TableView from "./TableView";
import store from "../store";
import { Provider } from "react-redux";
import { MemoryRouter } from "react-router";
import { loadDatabase } from "../utilities/db.utils";

beforeAll(async () => {
  await loadDatabase(store);
});

it("does not explode when rendered", async () => {
  const displayedCocktails = [store.getState().db.cocktails[0]];

  const tree = renderer.create(
    <Provider store={store}>
      <MemoryRouter>
        <TableView displayedCocktails={displayedCocktails} />
      </MemoryRouter>
    </Provider>
  );

  expect(tree).toMatchSnapshot();
});
