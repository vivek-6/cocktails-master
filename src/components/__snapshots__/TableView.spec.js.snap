// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`does not explode when rendered 1`] = `
<div
  className="MuiTableContainer-root"
>
  <table
    className="MuiTable-root"
    role={null}
  >
    <thead
      className="MuiTableHead-root"
      role={null}
    >
      <tr
        className="MuiTableRow-root MuiTableRow-head"
        role={null}
      >
        <th
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-head"
          scope="col"
        >
          Appearance
        </th>
        <th
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-head"
          scope="col"
        >
          Name
        </th>
        <th
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight"
          scope="col"
        >
          Actions
        </th>
      </tr>
    </thead>
    <tbody
      className="MuiTableBody-root"
      role={null}
    >
      <tr
        className="MuiTableRow-root MuiTableRow-hover"
        role={null}
        tabIndex={-1}
      >
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body"
        >
          <div
            aria-label="Recipe"
            className="MuiAvatar-root MuiAvatar-circle MuiAvatar-colorDefault"
            style={
              Object {
                "background": "linear-gradient(#D88317)",
                "backgroundColor": "#D88317",
              }
            }
          >
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <svg>
                Martini.svg
              </svg>
            </svg>
          </div>
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body"
        >
          Vesper
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
        >
          <button
            className="MuiButtonBase-root MuiButton-root MuiButton-text CocktailActions-button-2 MuiButton-textSecondary MuiButton-textSizeLarge MuiButton-sizeLarge"
            disabled={false}
            onBlur={[Function]}
            onClick={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={0}
            type="button"
          >
            <span
              className="MuiButton-label"
            >
              <svg
                aria-hidden={true}
                className="MuiSvgIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M16.5 3c-1.74 0-3.41.81-4.5 2.09C10.91 3.81 9.24 3 7.5 3 4.42 3 2 5.42 2 8.5c0 3.78 3.4 6.86 8.55 11.54L12 21.35l1.45-1.32C18.6 15.36 22 12.28 22 8.5 22 5.42 19.58 3 16.5 3zm-4.4 15.55l-.1.1-.1-.1C7.14 14.24 4 11.39 4 8.5 4 6.5 5.5 5 7.5 5c1.54 0 3.04.99 3.57 2.36h1.87C13.46 5.99 14.96 5 16.5 5c2 0 3.5 1.5 3.5 3.5 0 2.89-3.14 5.74-7.9 10.05z"
                />
              </svg>
            </span>
            <span
              className="MuiTouchRipple-root"
            />
          </button>
          <a
            aria-disabled={false}
            className="MuiButtonBase-root MuiButton-root MuiButton-text CocktailActions-button-2 MuiButton-textSecondary MuiButton-textSizeLarge MuiButton-sizeLarge"
            href="/cocktails/vesper"
            onBlur={[Function]}
            onClick={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            role="button"
            tabIndex={0}
          >
            <span
              className="MuiButton-label"
            >
              <svg
                aria-hidden={true}
                className="MuiSvgIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M21 5c-1.11-.35-2.33-.5-3.5-.5-1.95 0-4.05.4-5.5 1.5-1.45-1.1-3.55-1.5-5.5-1.5S2.45 4.9 1 6v14.65c0 .25.25.5.5.5.1 0 .15-.05.25-.05C3.1 20.45 5.05 20 6.5 20c1.95 0 4.05.4 5.5 1.5 1.35-.85 3.8-1.5 5.5-1.5 1.65 0 3.35.3 4.75 1.05.1.05.15.05.25.05.25 0 .5-.25.5-.5V6c-.6-.45-1.25-.75-2-1zm0 13.5c-1.1-.35-2.3-.5-3.5-.5-1.7 0-4.15.65-5.5 1.5V8c1.35-.85 3.8-1.5 5.5-1.5 1.2 0 2.4.15 3.5.5v11.5z"
                />
                <path
                  d="M17.5 10.5c.88 0 1.73.09 2.5.26V9.24c-.79-.15-1.64-.24-2.5-.24-1.7 0-3.24.29-4.5.83v1.66c1.13-.64 2.7-.99 4.5-.99zM13 12.49v1.66c1.13-.64 2.7-.99 4.5-.99.88 0 1.73.09 2.5.26V11.9c-.79-.15-1.64-.24-2.5-.24-1.7 0-3.24.3-4.5.83zM17.5 14.33c-1.7 0-3.24.29-4.5.83v1.66c1.13-.64 2.7-.99 4.5-.99.88 0 1.73.09 2.5.26v-1.52c-.79-.16-1.64-.24-2.5-.24z"
                />
              </svg>
            </span>
            <span
              className="MuiTouchRipple-root"
            />
          </a>
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
